code/game/objects/items/forged_weapons.dm

```diff
@@ -0,0 +1,343 @@
+
+	if(iswallturf(target) && proximity_flag)
+		var/turf/closed/wall/W = target
+		var/chance = (force_wielded + W.hardness * 0.5)//>lower hardness = stronger wall
+		if(chance < 10)
+			return FALSE
+
+		if(prob(chance))
+			playsound(src, 'sound/misc/bustconcrete1.ogg', 100, 1) //wonder where this is from!
+			W.dismantle_wall(TRUE)
+
+		else
+			playsound(src, 'sound/misc/bustconcrete2.ogg', 50, 1)
+			W.add_dent(WALL_DENT_HIT)
+			visible_message("<span class='danger'>[user] has smashed [W] with [src]!</span>", null, COMBAT_MESSAGE_RANGE)
+	return TRUE
+
+
+/obj/projectile/bullet/forged
+	damage = 0
+	hitsound_wall = "ricochet"
+	impact_effect_type = /obj/effect/temp_visual/impact_effect
+	var/datum/reagent/reagent_type
+	var/identifier = FORGED_BULLET_CASING
+	speed = 0.8
+	var/list/special_traits
+	var/radioactive = FALSE
+	var/fire = FALSE
+
+
+/obj/projectile/bullet/forged/proc/assign_properties(datum/reagent/reagent_type, caliber_multiplier)
+	if(reagent_type)
+		special_traits = list()
+		name = name += " ([reagent_type.name])"
+		color = reagent_type.color
+		damage = max(0.1, (reagent_type.density * 1.5 * caliber_multiplier))
+		speed = max(0, reagent_type.density / 2.5)
+		for(var/I in reagent_type.special_traits)
+			var/datum/special_trait/S = new I
+			LAZYADD(special_traits, S)
+			S.on_apply(src, identifier)
+
+
+/obj/projectile/bullet/forged/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	if(reagent_type)
+		for(var/I in special_traits)
+			var/datum/special_trait/A = I
+			if(prob(A.effectiveness))
+				A.on_hit(target, I = src, type = identifier)
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		var/limb_hit =  C.check_limb_hit(def_zone)
+		var/armour_block = C.getarmor(limb_hit, "bullet") * 0.01
+		if(!armour_block)
+			armour_block = 1
+		C.reagents.add_reagent(reagent_type.type, max(0, 1 * max(1, armour_block - armour_penetration)))
+		reagent_type.expose_mob(C, TOUCH, 1)
+
+//fuck tg
+/obj/projectile/bullet/forged/proc/open_flame(flame_heat=700)
+	var/turf/location = loc
+	if(ismob(location))
+		var/mob/M = location
+		var/success = FALSE
+		if(src == M.get_item_by_slot(ITEM_SLOT_MASK))
+			success = TRUE
+		if(success)
+			location = get_turf(M)
+	if(isturf(location))
+		location.hotspot_expose(flame_heat, 5)
+
+
+/obj/projectile/bullet/forged/Move()
+	. = ..()
+	if(!QDELETED(src))
+		for(var/I in special_traits)
+			var/datum/special_trait/A = I
+			if(prob(A.effectiveness))
+				A.on_hit(I = src, type = identifier)
+
+		if(radioactive)
+			radiation_pulse(src, 300)
+
+		if(fire)
+			open_flame()
+		var/turf/location = get_turf(src)
+		if(location && reagent_type)
+			reagent_type.expose_turf(location, 1)
+
+
+/obj/item/ammo_casing/forged
+	name = "forged bullet casing"
+	desc = "A custom bullet casing designed to be quickly changeable to any caliber. Utilizes, what else, bluespace technology and quantum entanglement to make every casing under you turn into the same caliber at once!"
```

Comment:
might be better to just not remark on bluespace stuff at all for that

Reply:
Alright. I am pretty sure the mention of it "changing all of them at once" is an artifact of a failed pr from hippie that just didn't get reverted? Kinda odd.

Reply:
I can't remember actually. Did I write this?

Reply:
Anyways, I just removed the second sentence entirely.